corehq/ex-submodules/phonelog/utils.py

```diff
@@ -70,3 +78,25 @@ def process_device_log(domain, xform):
             user_id=logged_in_user_id,
         ))
     DeviceReportEntry.objects.bulk_create(to_save)
+
+
+def _process_user_error_subreport(domain, xform):
+    errors = _get_logs(xform.form_data, 'user_error_subreport', 'user_error')
+    to_save = []
+    for i, error in enumerate(errors):
+        entry = UserErrorEntry(
+            domain=domain,
+            xform_id=xform.form_id,
+            i=i,
+            app_id=error['app_id'],
+            version_number=int(error['version']),
+            date=error["@date"],
+            server_date=xform.received_on,
+            user_id=error['user_id'],
+            expr=error['expr'],
+            msg=error['msg'],
+            session=error['session'],
+            type=error['type'],
+        )
+        to_save.append(entry)
+    UserErrorEntry.objects.bulk_create(to_save)
```

Comment:
guessing it's the same convention as the rest of the code, but curious whether it might make sense to try and make this atomic with the other models (may also already be solved by built in transaction management)